Support for detecting fixed-length prefix barcodes for fastq-multx

assaron opened the pull request:
Hi, in our protocol barcodes are the first 8 nucleotieds of the second read. I added support for this feature by adding '-k' option.

cyclospora commented:
Thanks for your submission, but it’s not clear that this patch covers the complete behavior that might be expected from the description of the new option, so we’re declining to merge it.  We might revisit the idea in the future however.

assaron commented:
Can you elaborate a bit, please? Sorry, it's probably not clear from the diff, but I just added an option to set `blen` manually (see the screenshot attached with diff that doesn't look at indents).
![2017-02-13-185921_1366x751_scrot](https://cloud.githubusercontent.com/assets/1040217/22890915/c547244e-f21e-11e6-951e-de14d892abab.png)

 I'm 100% sure that it doesn't break any previous behaviour and if some unexpected behaviour will come out when using this option, I guess it could reported and fixed as a normal bug.

assaron commented:
Hi, any updates here? I'm still a little confused why the pull request was rejected. How can I change it to make it into upstream?

Previously, In the code blen parameter was determined automatically. I added an option to provide it manually. Usually it's a good thing to make parameters be set manually in addition to automatic setting. If you find the option name misleading, I can change to whatever suits best.
